### ISSUE TITLE
Variable type info in statusbar

### ISSUE THREAD

**Issue by deltaskelta** [joefitzgerald/go-plus#700]
### Prerequisites

* [ ] Have you tried launching `atom .` from the terminal in your project's directory?
* [ ] Have you verified the output from `go env` is correct? If it is, please include the output in this issue.
* [x] Have you updated Atom to the latest version?
* [ ] Have you tried using [Atom Beta](https://atom.io/beta), which can be run side-by-side with Atom Stable? Is the behavior the same, or different?
* [x] Have you updated your Atom packages to the latest versions?
* [x] Have you read the [FAQ](https://github.com/joefitzgerald/go-plus/wiki/FAQ)?
* [x] Have you searched [the issues](https://github.com/joefitzgerald/go-plus/issues?utf8=%E2%9C%93&q=is%3Aissue) to see if others are experiencing the same issue?

### Description

I was having a look through the issues and I noticed #190, which as I can see was implemented, but it only seems to show function signatures and not variable types. I use vim-go in vim and this was probably my most relied upon feature that I miss. 

How hard would it be to include variable types in the status bar instead of only function signatures? I could have a go at implementing it if you think its possible with the current state of things. 

### Output from `atom -v && apm -v`

Atom    : 1.21.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
apm  1.18.5
npm  3.10.10
node 6.9.5 x64
python 2.7.10
git 2.14.2

### Steps to Reproduce

```go
func main() {
    var i int
    log.Println(i)
    i += 2
}
```
### Expected Behavior

Hover on `i` both in the call to `Println` and after that, and have the type of `i` show in the status bar.

### Actual Behavior

Hovering over the `i` in the call to `Println` showed the function signature for `Println` and hovering over `i` below that shows nothing

**Comment by zmb3** [comment]
This seems to be a duplicate of #542. 

Note that we don’t have hover, but you can put your cursor over anything and run the golang:showdoc command (alt+d by default) to get info on it.

**Comment by deltaskelta** [comment]
I do not think that issue is the same as I am suggesting. My use of the word hover was meant to mean that the cursor was hovering on the variable, not the mouse (I hope that makes it clearer)

Presently, if I put the cursor on `log.Println`, it will show the function signature in the status bar. I am suggesting adding the case when the cursor is on the variable `i`, it will just say `int` in the status bar

**Comment by zmb3** [comment]
Ah, I see. You’re correct. We could fix this pretty easily by using
gogetdoc instead of gocode to provide the status bar info.

**Comment by deltaskelta** [comment]
Would that involve changing the current function signature in the status bar to use gogetdoc as well? 

Would `gocode` be completely eliminated from the plugin or is it used throughout the plugin in different places as well?

**Comment by zmb3** [comment]
Gocode is also used for autocomplete, that part would remain unchanged.

**Comment by deltaskelta** [comment]
Ok, I am a little bit confused. I was poking through the code and commits and I noticed that when this feature was added, it looks like it comes from [wndhydrnt/go-signature-statusbar](https://github.com/wndhydrnt/go-signature-statusbar) so wouldn't that be the place to think about adding the functionality?

Should it be that the `gogetdoc` is made into something like `gocodeprovider` and then the functionality is actually added in the `go-signature-statusbar` project?
